src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -75,7 +75,13 @@ public class StateManager {
     private final ObservableList<String> searchHistory = FXCollections.observableArrayList();
 
     public StateManager() {
-        activeGroups.bind(Bindings.valueAt(selectedGroups, activeDatabase.orElseOpt(null).map(BibDatabaseContext::getUid)));
```

Comment:
Hasn't JavaFX the option to have "Opt..."? The `NULL_CONTEXT` is a nice idea, but we never use that in JabRef. We have Optionals.

Reply:
I was running into issues when returning `activeGroups.readOnlyProperty()` where it was returning out a `null` and causing problems. Upon review, I think I can achieve the same effect with the following:
```
    public StateManager() {
        activeGroups.bind(Bindings.valueAt(
            selectedGroups,
            activeDatabase
                .orElseOpt(null)
                .map(BibDatabaseContext::getUid))
            .orElse(FXCollections.observableArrayList()));
    }
```